File: test-src/12_inheritance.kt
open class Base {
  open fun v() { println("ERROR-Open") }
  fun nv() { println("4") }
  init { println("1") }
}

open class Derived() : Base() {
  override open fun v() { println("ERROR-Derived") }
  fun x() { println("6") }
  init { println("2") }
}

class Derived2 : Derived() {
  override fun v() { println("5") }
  fun y() { println("7") }
  init { println("3") }
}

fun main(args: Array<String>) {
  val x = Derived2()
  x.nv()
  x.v()
  x.x()
  x.y()
}

File: dest/12_inheritance.swift
public class Base {
  func v() { print("ERROR-Open") }
  func nv() { print("4") }
  init() { print("1") }
}

public class Derived: Base {
  override func v() { print("ERROR-Derived") }
  func x() { print("6") }
  override init() { print("2") }
}

public class Derived2: Derived {
  override func v() { print("5") }
  func y() { print("7") }
  override init() { print("3") }
}

func main(args: [String]) {
  let x = Derived2()
  x.nv()
  x.v()
  x.x()
  x.y()
}

main([])
